parametric/test_span_sampling.py

```diff
@@ -442,3 +442,88 @@ def test_multi_rule_independent_rate_limiters_sss013(test_agent, test_library):
             pass
     span = find_span_in_traces(test_agent.wait_for_num_traces(1), Span(name="web.request"))
     assert span["metrics"].get(SINGLE_SPAN_SAMPLING_MAX_PER_SEC) == 1
+
+@pytest.mark.skip_library("nodejs", "Not implemented")
+@pytest.mark.skip_library("python", "RPC issue causing test to hang")
+@pytest.mark.parametrize(
+    "library_env",
+    [
+        {
+            "DD_SPAN_SAMPLING_RULES": json.dumps(
+                [{"service": "webserver", "name": "parent", "sample_rate": 1.0, "max_per_second": 50}]
+            ),
+           "DD_TRACE_SAMPLE_RATE": 0,
+        }
+    ],
+)
+def test_root_span_selected_by_sss014(test_agent, test_library):
+    """Single spans selected by SSS must be kept and shouldn't affect child span sampling priority.
+    
+    We're essentially testing to make sure that the span sampling rule keeps selected spans regardless of the trace sampling decision
+    and doesn't affect child spans that are dropped by the tracer sampling mechanism.
+    """
+    with test_library:
+        with test_library.start_span(name="parent", service="webserver") as parent_span:
+            with test_library.start_span(name="child", service="webserver", parent_id=parent_span.span_id):
+                pass
+
+    traces = test_agent.wait_for_num_traces(1, clear=True)
+
+    parent_span = find_span_in_traces(traces, Span(name="parent", service="webserver"))
+    child_span = find_span_in_traces(traces, Span(name="child", service="webserver"))
+
+    # root span should be kept by defined the SSS rules
+    # don't check SAMPLING_PRIORITY_KEY of the parent since it could be set to drop,
+    # but it isn't a requirement
+    assert parent_span["metrics"].get(SINGLE_SPAN_SAMPLING_RATE) == 1.0
+    assert parent_span["metrics"].get(SINGLE_SPAN_SAMPLING_MECHANISM) == SINGLE_SPAN_SAMPLING_MECHANISM_VALUE
+    assert parent_span["metrics"].get(SINGLE_SPAN_SAMPLING_MAX_PER_SEC) == 50
+
+    # child span should be dropped by defined trace sampling rules
+    assert child_span["metrics"].get(SAMPLING_PRIORITY_KEY) == -1
```

Comment:
Is this assertion correct? I see no reference to sampling priority key in the RFC other than setting to 2 on user-keep. This is the single failure remaining for Node.js. If I comment this out the rest pass.

Reply:
If we remove this assertion then Java Tracer prior to the fix will pass this test.

Reply:
@dgoffredo ^

Reply:
I think that the correct assertion would be:
```python
assert parent_span["metrics"].get(SAMPLING_PRIORITY_KEY) == -1
```
Sampling priority is about the trace. We convey the sampling priority of the trace (chunk) by setting the sampling priority on the local root span.

What we want to ensure is that the trace is being dropped despite the span sampling at play. So, we should check the sampling priority of the local root span, and make sure it indicates "drop." The Agent will then keep any spans that have `_dd.span_sampling.mechanism` set.

Reply:
Ref: https://github.com/DataDog/datadog-agent/blob/0b2ce0478d726be5642a666e75dada49403c0ccd/pkg/trace/agent/agent.go#L260-L262

Reply:
@dgoffredo I think you're right! I'm going to clean up this part in Java Tracer. Thanks for explanation!

Reply:
Cleaned this logic in Java Tracer in [this PR](https://github.com/DataDog/dd-trace-java/pull/4695)